src/model.js

```diff
@@ -0,0 +1,242 @@
+const yelp = require("yelp-fusion");
+const proj4 = require("proj4");
+const _ = require("lodash");
+const async = require("async");
+
+const {
+  koopProviderYelp: { api_key },
+} = require("config");
+const client = yelp.client(api_key);
+const YELP_API_MAX_RADIUS = 40000;
+
+const proj = proj4("GOOGLE", "WGS84");
+
+function Model() {}
+
+// each model should have a getData() function to fetch the geo data
+// and format it into a geojson
+Model.prototype.getData = function (req, callback) {
+  if (req.query && req.query.returnCountOnly) {
+    return callback(null, {
+      type: "FeatureCollection",
+      count: 2001,
+      features: [],
+    });
+  }
+  const queries = buildQueries(req.query);
+  const featureCollection = {
+    type: "FeatureCollection",
+    features: [],
+  };
+
+  async.each(queries, search, finish);
+
+  function search(query, cb) {
+    // If this service is added to an ArcGIS Online Map, usually it makes multiple
+    // calls (4 quadrants). When you make 4 calls to the Yelp API at the same
+    // time, you get a Yelp error, so for now set a random pause 0.5-2 seconds.
+    // TODO: Remove this random pause code, and change it so that if we get a
+    // "TOO_MANY_REQUESTS_PER_SECOND" error, retry after a random pause.
+
+    const randomPauseTime = Math.random() * (2500 - 500) + 500;
+
+    setTimeout(() => {
+      searchYelp(query, function (err, features) {
+        if (err) return cb(err);
+        featureCollection.features = featureCollection.features.concat(
+          features
+        );
+        cb();
+      });
+    }, randomPauseTime);
+  }
+
+  function finish(err) {
+    // Service metadata
+    featureCollection.metadata = {
+      name: "Yelp",
+      description: `Generated from the Yelp API.`,
+    };
+
+    if (err) {
+      callback(err);
+    } else {
+      callback(null, featureCollection);
+    }
+  }
+};
+
+// Wrap the call to Yelp, this will make testing easier and decouple us from the specific client lib
+function searchYelp(query, callback) {
+  client.search(query).then(
+    function (rawResponse) {
+      // console.log("rawResponse", rawResponse);
+      const features = translate(rawResponse.jsonBody, query);
+      console.log("Search returned", features.length, "results");
+      callback(null, features);
+    },
+    function (err) {
+      // console.log("err", err);
+      callback(err);
+    }
+  );
+}
+
+function buildQueries(options) {
+  let queries;
+  if (options.geometry) {
+    queries = [buildQuery(options, options.geometry)];
+  } else {
+    queries = [buildQuery(options)];
```

Comment:
No need to use `queries` variable here or `else`.  Just use `return`, i.e.,
```
if (options.geometry) {
  return [buildQuery(options, options.geometry)];
} 
```

Reply:
Yeah, the multiple queries was a hold-over from the previous code, I forgot to remove this. thanks.